### DISCUSSION TITLE
validating claim set of decoded token

### DISCUSSION THREAD

**Discussion by ibash** [panva/jose#667] [Q&A]
I want to do something like this in the frontend of our codebase:
```javascript
const payload = jose.decodeJwt(token)
const isValid = jose.validateClaimSet(payload)
```

At this part of the code I'm not interested in verifying the signature for security reasons (that's done elsewhere). Instead the use case is a quick check to make sure the token isn't expired / likely to be invalid.

I'm not using jwtVerify because I need this code to be synchronous.

I looked for a way to do this in the code but afaict it's not supported right now. Is that right or did I miss something?

**Comment by panva** [comment]
You didn't miss anything. A JWT is evaluated as a whole including the signature with the jwtVerify API.

**Reply by ibash** [reply]
jfc you reply fast. Okay, thank you!
